tmva/tmva/src/DNN/Architectures/Cpu/ActivationFunctions.hxx

```diff
@@ -79,7 +79,7 @@ void TCpu<AFloat>::ReluDerivative(TCpuTensor<AFloat> & B,
 
 //______________________________________________________________________________
 template<typename AFloat>
-void TCpu<AFloat>::Sigmoid(TCpuTensor<AFloat> & B)
+void TCpu<AFloat>::Sigmoid(Tensor_t & B)
```

Comment:
I would not do this change and keep as the other functions

Reply:
The motivation here was to make the declaration match with the definition, to avoid this warning:

```
/home/sftnight/build/workspace/root-makedoc-master/rootspi/rdoc/src/master/tmva/tmva/src/DNN/Architectures/Cpu/ActivationFunctions.hxx:82: warning: no uniquely matching class member found for 
  template < AFloat >
  void TMVA::DNN::TCpu< AFloat >::Sigmoid(TMVA::DNN::TCpuTensor< AFloat > &B)
Possible candidates:
  'static void TMVA::DNN::TCpu< AReal >::Sigmoid(Tensor_t &B)' at line 305 of file /home/sftnight/build/workspace/root-makedoc-master/rootspi/rdoc/src/master/tmva/tmva/inc/TMVA/DNN/Architectures/Cpu.h
  'static void TMVA::DNN::TCpu< AReal >::Sigmoid(Matrix_t &YHat, const Matrix_t &)' at line 378 of file /home/sftnight/build/workspace/root-makedoc-master/rootspi/rdoc/src/master/tmva/tmva/inc/TMVA/DNN/Architectures/Cpu.h
```

I totally agree that it's not necessary, but doxygen generates now a bazillion warnings https://lcgapp-services.cern.ch/root-jenkins/view/ROOT/job/root-makedoc-master/2590/consoleText, so removing them would help to keep the documentation up to date in the future, so that real true warnings can be detected. Or if we believe there is a bug in doxygen, we could try to submit a bug report to them.

Reply:
I agree that it should be consistent, no matter what doxygen says. It's also better for manual code reading.

I think using just `Tensor_t` is a bit confusing. because there is actually some information lost to the reader of the code: it's not clear that this is a templated type anymore, because it's a member of the templated class `TCpu<AFloat>`.

What about `TCuda<AFloat>::Tensor_t`? That would be consistent and tell the full story. What does doxygen say about that?

Reply:
Not sure about what doxygen says, at the moment I do not have this branch 'built' anymore locally. It would be great to have this inplace at some point: https://github.com/root-project/root/issues/9953
and/or https://github.com/root-project/jenkins-pipelines/issues/10

Reply:
That's okay, I don't think it's worth it so set up the whole pipeline locally. But we can check the log in Jenkins, right?

Maybe to save time, we can merge the PR with the fully qualified names now, and then if we see there are still warnings in doxygen in the official build we follow up?

Reply:
Yep, we can check the log in jenkins, that's right, though it's a bit cluttered due to (https://github.com/root-project/jenkins-pipelines/issues/10).
Sounds like a plan!